test/unit/lib/aws/awssigv4signer.test.js

```diff
@@ -77,20 +77,44 @@ test('Sign with SigV4 failure (with empty region)', (t) => {
   }
 });
 
-test('Sign with SigV4 failure (without getCredentials function)', (t) => {
```

Comment:
I think you want both tests, without getCredentials and with default.

Reply:
the default is without getCredentials, the others test use custom provided getCredentials function